File: src/com/xu/music/player/player/Player.java
package com.xu.music.player.player;

import javax.sound.sampled.AudioFormat;
import javax.sound.sampled.AudioFormat.Encoding;
import javax.sound.sampled.AudioInputStream;
import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.net.URL;

public interface Player {

    /**
     * Java Music 加载音频
     *
     * @param url 音频文件url地址
     * @return boolean 是否加载成功
     * @Title: load
     * @Description: Java Music 加载音频
     * @date 2019年10月31日19:06:39
     */
    void load(URL url);

    /**
     * Java Music 加载音频
     *
     * @param file 音频文件
     * @return boolean 是否加载成功
     * @Title: load
     * @Description: Java Music 加载音频
     * @date 2019年10月31日19:06:39
     */
    void load(File file);

    /**
     * Java Music 加载音频
     *
     * @return boolean 是否加载成功
     * @Title: load
     * @Description: Java Music 加载音频
     * @date 2019年10月31日19:06:39
     */
    void load(String path);

    /**
     * Java Music 加载音频
     *
     * @param stream 音频文件输入流
     * @return boolean 是否加载成功
     * @Title: load
     * @Description: Java Music 加载音频
     * @date 2019年10月31日19:06:39
     */
    void load(InputStream stream);

    /**
     * Java Music 加载音频
     *
     * @param encoding Encoding
     * @param stream   AudioInputStream
     * @return boolean 是否加载成功
     * @Title: load
     * @Description: Java Music 加载音频
     * @date 2019年10月31日19:06:39
     */
    void load(Encoding encoding, AudioInputStream stream);

    /**
     * Java Music 加载音频
     *
     * @param stream AudioFormat
     * @param stream AudioInputStream
     * @return boolean 是否加载成功
     * @Title: load
     * @Description: Java Music 加载音频
     * @date 2019年10月31日19:06:39
     */
    void load(AudioFormat format, AudioInputStream stream);

    /**
     * Java Music 结束播放
     *
     * @throws IOException
     * @Title: end
     * @Description: Java Music 结束播放
     * @date 2019年10月31日19:06:39
     */
    void end() throws IOException;

    /**
     * Java Music 暂停播放
     *
     * @Title: stop
     * @Description: Java Music 结束播放
     * @date 2019年10月31日19:06:39
     */
    void stop();

    /**
     * Java Music 开始播放
     *
     * @throws Exception
     * @Title: start
     * @Description: Java Music 结束播放
     * @date 2019年10月31日19:06:39
     */
    void start() throws Exception;

    /**
     * Java Music 是否打开
     *
     * @return 是否打开
     * @Title: isOpen
     * @Description: Java Music 是否打开
     * @date 2019年10月31日19:06:39
     */
    boolean isOpen();

    /**
     * Java Music 是否运行
     *
     * @return 是否运行
     * @Title: isAlive
     * @Description: Java Music 是否运行
     * @date 2019年10月31日19:06:39
     */
    boolean isAlive();

    /**
     * Java Music 是否正在播放
     *
     * @return 是否正在播放
     * @Title: isRuning
     * @Description: Java Music 是否正在播放
     * @date 2019年10月31日19:06:39
     */
    boolean isRuning();

    /**
     * Java Music 音频信息
     *
     * @return 音频信息
     * @Title: info
     * @Description: Java Music 音频信息
     * @date 2019年10月31日19:06:39
     */
    String info();

    /**
     * Java Music 音频播放时长
     *
     * @return 音频播放时长
     * @Title: length
     * @Description: Java Music 音频播放时长
     * @date 2019年10月31日19:06:39
     */
    double length();

}

File: src/main/kotlin/com/xu/music/player/player/Player.kt
package com.xu.music.player.player

import java.io.File
import java.net.URL
import javax.sound.sampled.AudioFormat
import javax.sound.sampled.AudioInputStream

/**
 * 音频播放
 *
 * @date 2024年6月4日19点07分
 * @since SWT-V1.0.0.0
 */
interface Player {

    /**
     * 加载音频
     *
     * @param url 音频文件url
     * @throws Exception 异常
     * @date 2019年10月31日19:06:39
     */
    @Throws(Exception::class)
    fun load(url: URL?)

    /**
     * 加载音频
     *
     * @param file 音频文件
     * @throws Exception 异常
     * @date 2019年10月31日19:06:39
     */
    @Throws(Exception::class)
    fun load(file: File)

    /**
     * 加载音频
     *
     * @param path 文件路径
     * @throws Exception 异常
     * @date 2019年10月31日19:06:39
     */
    @Throws(Exception::class)
    fun load(path: String?)

    /**
     * 加载音频
     *
     * @param stream 音频文件输入流
     * @throws Exception 异常
     * @date 2019年10月31日19:06:39
     */
    @Throws(Exception::class)
    fun load(stream: AudioInputStream)

    /**
     * 加载音频
     *
     * @param encoding Encoding
     * @param stream   AudioInputStream
     * @throws Exception 异常
     * @date 2019年10月31日19:06:39
     */
    @Throws(Exception::class)
    fun load(encoding: AudioFormat.Encoding?, stream: AudioInputStream?)

    /**
     * 加载音频
     *
     * @param format AudioFormat
     * @param stream AudioInputStream
     * @throws Exception 异常
     * @date 2019年10月31日19:06:39
     */
    @Throws(Exception::class)
    fun load(format: AudioFormat?, stream: AudioInputStream?)

    /**
     * 暂停播放
     *
     * @date 2019年10月31日19:06:39
     */
    fun pause()

    /**
     * 继续播放
     *
     * @param duration 音频位置
     * @date 2019年10月31日19:06:39
     */
    fun resume(duration: Long)

    /**
     * 开始播放
     *
     * @date 2019年10月31日19:06:39
     */
    fun play()

    /**
     * 结束播放
     *
     * @date 2019年10月31日19:06:39
     */
    fun stop()

    /**
     * 设置音量
     *
     * @param volume 音量
     * @date 2019年10月31日19:06:39
     */
    fun volume(volume: Float)

    /**
     * 获取音频播放位置
     *
     * @return 播放位置
     * @date 2019年10月31日19:06:39
     */
    fun position(): Double

    /**
     * 获取音频总时长
     *
     * @return 音频总时长
     * @date 2019年10月31日19:06:39
     */
    fun duration(): Double

    /**
     * 是否正在播放
     *
     * @return 是否正在播放
     * @date 2019年10月31日19:06:39
     */
    fun playing(): Boolean

    /**
     * 是否正在
     *
     * @return 是否正在播放
     * @date 2019年10月31日19:06:39
     */
    fun pausing(): Boolean

}
